samples/net/mqtt_publisher/src/main.c

```diff
@@ -11,6 +11,9 @@ LOG_MODULE_REGISTER(net_mqtt_publisher_sample, LOG_LEVEL_DBG);
 #include <zephyr/net/socket.h>
 #include <zephyr/net/mqtt.h>
 #include <zephyr/random/random.h>
+#if IS_ENABLED(CONFIG_LOG_BACKEND_MQTT)
```

Comment:
Just use plain `#if defined(CONFIG_LOG_BACKEND_MQTT)`, IS_ENABLED() isn't really needed in such case.